src/table/utils/selections-utils.js

```diff
@@ -98,41 +103,54 @@ export const getSelectedRows = ({ selectedRows, cell, evt }) => {
 
 const selectCell = (state, payload) => {
   const { api, rows, colIdx } = state;
-  const { cell, announce, evt } = payload;
+  const { cell, announce, evt, isSelectMultiValues } = payload;
   let selectedRows = {};
 
-  if (colIdx === -1) {
-    api.begin('/qHyperCubeDef');
-  } else if (colIdx === cell.colIdx) {
-    selectedRows = { ...rows };
-  } else {
-    return state;
-  }
+  if (colIdx === -1) api.begin('/qHyperCubeDef');
+  else if (colIdx === cell.colIdx) selectedRows = { ...rows };
+  else return state;
 
   selectedRows = getSelectedRows({ selectedRows, cell, evt });
   const selectedRowsLength = Object.keys(selectedRows).length;
   handleAnnounceSelectionStatus({
     announce,
     rowsLength: selectedRowsLength,
-    isAddition: selectedRowsLength > Object.keys(rows).length,
+    isAddition: selectedRowsLength >= Object.keys(rows).length,
   });
 
   if (selectedRowsLength) {
-    api.select({
-      method: 'selectHyperCubeCells',
-      params: ['/qHyperCubeDef', Object.values(selectedRows), [cell.colIdx]],
-    });
+    !isSelectMultiValues &&
+      api.select({
+        method: 'selectHyperCubeCells',
+        params: ['/qHyperCubeDef', Object.values(selectedRows), [cell.colIdx]],
+      });
     return { ...state, rows: selectedRows, colIdx: cell.colIdx };
   }
 
   api.cancel();
   return { ...state, rows: selectedRows, colIdx: -1 };
 };
 
+const selectMultiValues = (state, payload) => {
+  const { api, rows } = state;
+  const { colIdx } = payload.cell;
+  const rowsLength = Object.keys(rows).length;
```

Comment:
I would say better to add a new prop indicating that you have values that need to be sent to backend. in your solution now you could be in selection mode, have some selections made, then just press and release shift and there will be a call to engine. We want to avoid that

Reply:
Really good point. Fixed.